jwtclient/client.go

```diff
@@ -0,0 +1,22 @@
+package jwtclient
+
+import (
+	"errors"
+	"net/http"
+)
+
+// NewJWTClient is http.Client decorator which returns a client with `JWTTransport` transport.
+// `JWTTransport` will wrap the request, generate JWT token and updated Authorization header.
+func NewJWTClient(client *http.Client, opts ...Option) (*http.Client, error) {
```

Comment:
that makes sense. I recall we discussed with @alberts we want an http.Client decorator, but a custom transport also works in our case. Fixed